File: spavanac.py
#Jacob Pawlak
#February 14th 2017 Happy Valentines Day
#Spavanac
#https://open.kattis.com/problems/spavanac

def main():

	time_str = input()
	time_list = time_str.split()
	hours = int(time_list[0])
	mins = int(time_list[1])
	mins = mins - 45
	if(mins < 0):
		mins += 60
		hours -= 1
		if(hours < 0):
			hours = 23
	print(hours, mins)


main()

File: spavanac.cpp
//Jacob Pawlak
//February 14th, 2017 Happy Valentines day!
//Spavanac
//https://open.kattis.com/problems/spavanac

#include<iostream>

using namespace std;

int main(){

	int hours, mins = 0;
	cin >> hours >> mins;
	mins -= 45;
	if(mins < 0){
		mins += 60;
		hours--;
		if(hours < 0)
			hours = 23;

	}

	cout << hours << " " << mins << endl;
	return 0;
}
